Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -74,16 +73,24 @@ void AllocatorBase::PostCreate()
         }
     }
 
-#if PLATFORM_MEMORY_INSTRUMENTATION_ENABLED
-    m_platformMemoryInstrumentationGroupId = AZ::PlatformMemoryInstrumentation::GetNextGroupId();
-    AZ::PlatformMemoryInstrumentation::RegisterGroup(m_platformMemoryInstrumentationGroupId, GetDescription(), AZ::PlatformMemoryInstrumentation::m_groupRoot);
-#endif
+    const auto debugConfig = GetDebugConfig();
+    if (!debugConfig.m_excludeFromDebugging)
+    {
+        SetRecords(aznew Debug::AllocationRecords((unsigned char)debugConfig.m_stackRecordLevels, debugConfig.m_usesMemoryGuards, debugConfig.m_marksUnallocatedMemory, GetName()));
```

Comment:
What if we updated the m_records member to be an `AZStd::unique_ptr<Debug::AllocationRecords>` and `SetRecords` to accept a `unique_ptr<Debug::AllocationRecords>` by value.

This call can remain the same, while the the PreDestroy call just be reduced to
```
SetRecords(nullptr);`
```

Reply:
I can do that as part of another change, didnt want to affect it too much